File: www/user_form.py
#!/usr/bin/python3

import cgi
import os
import sys
from urllib.parse import unquote
import cgitb
cgitb.enable()

request_method = os.environ.get("REQUEST_METHOD").upper()
route = unquote("." + os.environ.get("ROUTE"))

def output_html(custom_section):
    html_content = f'''
        <!DOCTYPE html>
        <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1.0">
            <title>User Form</title>
        </head>
        <body>
            {custom_section}
        </body>
        </html>
        '''
	
    print(html_content)


if request_method == "GET":
	form = cgi.FieldStorage()
	
	if not form.list:
		output_html('''<h2>Enter Your Info</h2>
        <form action="/user_form.py" method="get">
            <input type="text" name="name" id="name" placeholder="Full Name"><br>
            <input type="number" name="age" id="age" placeholder="Age"><br>
            <input type="submit">
        </form>''')
	else:
		name_param = form.getvalue('name')
		age_param  = form.getvalue('age')
		output_html(f'''
            <h1>Hello! {name_param}, age {age_param}</h1>
        ''')

File: www/user_form.php
#!/usr/bin/php-cgi
<?php
if ($_SERVER["REQUEST_METHOD"] === "POST") {
    // POST request: Display the submitted data
    $name = $_POST["name"];
    $age = $_POST["age"];
    echo "<html><body>";
    echo "<h1>Submitted Data</h1>";
    echo "<p>Name: $name</p>";
    echo "<p>Age: $age</p>";
    echo "</body></html>";
} else {
    // GET request: Display the input form
    echo "<html><body>";
    echo "<h1>Enter Your Information</h1>";
    echo '<form method="post" action="' . $_SERVER["PHP_SELF"] . '">';
    echo '  <label for="name">Name:</label>';
    echo '  <input type="text" id="name" name="name" required><br>';
    echo '  <label for="age">Age:</label>';
    echo '  <input type="number" id="age" name="age" required><br>';
    echo '  <input type="submit" value="Submit">';
    echo '</form>';
    echo "</body></html>";
}
?>